CONTRIBUTING.md

```diff
@@ -153,3 +147,30 @@ There is a [High-level code walkthrough](doc/code-walkthrough.md) describing the
 ### Logos
 
 We have [documentation for logo usage](doc/logos.md) which includes [contribution guidance](doc/logos.md#contributing-logos)
+
+## Pull Requests
+
+All code changes are incorporated via pull requests, and pull requests are always squashed into a single commit on merging. Therefore there's no requirement to squash commits within your PR, but feel free to squash or restructure the commits on your PR branch if you think it will be helpful. PRs with well structured commits are always easier to review!
+
+Because all changes are pulled into the main branch via squash merges from PRs, we do **not** support overwriting any aspects of the git history once it hits our main branch. Notably this means we do not support amending commit messages, nor adjusting commit author information once merged.
```

Comment:
I guess the other case where we might want to (or someone might request that we) rewrite history is where credentials have been exposed. Hopefully we'd catch it in review, but lets say we commit a token or password to the repo. Obviously the ideal course of action is we revoke the token or change the password and make a subsequent commit that removes the credentials. Thinking about that example, can we see a situation in which we would want to break our own rule here?

Reply:
Fair question. I think my preference is to set a pretty categorical rule in documentation vs trying to enumerate a full set of exceptions or having a softer policy with vague exceptional cases. We will always have to the option to evaluate the specific scenario should it ever come up again, but for the docs targeted to the audience of contributors, I really want to stress that it's one of their responsibilities.